src/components/resume-builder/components/edit-panel/WorkExperienceEditBox.tsx

```diff
@@ -0,0 +1,241 @@
+          index={expIndex}
+          data={exp}
+          addNewDescription={addNewDescription}
+          deleteDescription={deleteDescription}
+          deleteExperience={deleteExperience}
+          handleDescriptionChange={handleDescriptionChange}
+          handleExperienceChange={handleExperienceChange}
+        />
+      ))}
+
+      <BlueButton label="Add New Experience" onClick={addNewExperience} />
+    </div>
+  );
+};
+
+interface ExperienceEditBoxProps {
+  index: number;
+  data: Experience;
+  handleExperienceChange: (
+    index: number,
+    keyName: keyof Experience,
+    value: string,
+  ) => void;
+  deleteExperience: (index: number) => void;
+  handleDescriptionChange: (
+    expIndex: number,
+    descIndex: number,
+    value: string,
+  ) => void;
+  addNewDescription: (index: number) => void;
+  deleteDescription: (expIndex: number, descIndex: number) => void;
+}
+
+const ExperienceEditBox: React.FC<ExperienceEditBoxProps> = ({
+  index,
+  data,
+  handleExperienceChange,
+  deleteExperience,
+  handleDescriptionChange,
+  addNewDescription,
+  deleteDescription,
+}) => {
+  return (
+    <div className="p-1 border rounded relative flex flex-col gap-1">
+      <div className="flex flex-row items-center justify-between">
+        <p className="text-xs font-medium">{`Experience #${index}`}</p>
+        <ButtonWithCrossIcon onClick={() => deleteExperience(index)} />
+      </div>
+      <InputField
+        value={data.companyName}
+        onChange={(event) =>
+          handleExperienceChange(index, 'companyName', event.target.value)
+        }
+        placeholder="Company Name"
+      />
+      <InputField
+        type="text"
+        value={data.jobTitle}
+        onChange={(event) =>
+          handleExperienceChange(index, 'jobTitle', event.target.value)
+        }
+        placeholder="Job Title"
+      />
+      <div className="flex gap-2">
+        <InputField
+          type="text"
+          value={data.startDate}
+          onChange={(event) =>
+            handleExperienceChange(index, 'startDate', event.target.value)
+          }
+          placeholder="Start Date"
+        />
+        <InputField
+          type="text"
+          value={data.endDate}
+          onChange={(event) =>
+            handleExperienceChange(index, 'endDate', event.target.value)
+          }
+          placeholder="End Date"
+        />
+      </div>
+
+      {data.description.map((desc, descIndex) => (
+        <div key={descIndex} className="flex items-center gap-2">
+          <InputField
+            type="text"
+            value={desc}
+            onChange={(event) =>
+              handleDescriptionChange(index, descIndex, event.target.value)
+            }
+            placeholder={`Description #${descIndex}`}
+          />
+          <ButtonWithCrossIcon
+            onClick={() => deleteDescription(index, descIndex)}
+          />
+        </div>
+      ))}
+      <ButtonWithPlusIcon
```

Comment:
Check for the buttons in the similar components if right buttons are being used.